File: SurveyApplication.Domain/CauHoi.cs
﻿using System.ComponentModel.DataAnnotations;
using SurveyApplication.Domain.Common;

namespace SurveyApplication.Domain;

public class CauHoi : BaseDomainEntity
{
    [Required] [MaxLength(250)] public string MaCauHoi { get; set; }

    public short LoaiCauHoi { get; set; }

    [Required] public string TieuDe { get; set; }

    public string NoiDung { get; set; }
    public int SoLuongFileToiDa { get; set; }
    public int KichThuocFile { get; set; }
    public bool? IsOther { get; set; }
    public string LabelCauTraLoi { get; set; }
}

File: SurveyApplication.UI/src/app/models/CauHoi.ts
import { Base } from './Common/Base';
import { Cot, Hang } from './SurveyConfig';

export interface CauHoi extends Base {
  loaiCauHoi: number;
  maCauHoi: string;
  batBuoc: boolean | null;
  tieuDe: string;
  isOther: boolean | null;
  labelCauTraLoi: string;
  kichThuocFile: number;
  loaiCauHoiText: string;
}
export interface CreateUpdateCauHoi {
  id: number;
  maCauHoi: string;
  loaiCauHoi: number;
  batBuoc: boolean | null;
  tieuDe: string;
  noidung: string;
  soLuongFileToiDa: number | null;
  kichThuocFile: number | null;
  isOther: boolean | null;
  labelCauTraLoi: string;
  priority: number;
  lstCot: Cot[]; //Câu hỏi phụ
  lstHang: Hang[]; // Câu trả lời
}

export interface BangKhaoSatCauHoi {
  id: number | null;
  idCauHoi: number;
  priority: number | null;
  isRequired: boolean | null;
  panelTitle: string;
  maCauHoi: string | null;
  tieuDe: string | null;
}

export interface CauHoiGroup {
  panelTitle: string;
  bangKhaoSatCauHoi: BangKhaoSatCauHoi[];
}
